File: Cliente1/CuentaManagement.vb

Public Class CuentaManagement
    Private crudCuenta As AccesoDatos.Crud.CuentaCrudFactory
    Public Sub CreditoManagement()
        crudCuenta = New AccesoDatos.Crud.CuentaCrudFactory()
    End Sub

    Public Function Create(cuenta As Entidades.Cuenta)
        crudCuenta.Create(cuenta)
    End Function

    Public Function RetrieveAll() As List(Of Entidades.Cuenta)
        Return crudCuenta.RetrieveAll(Of Entidades.Cuenta)
    End Function

    Public Function RetrieveById(cuenta As Entidades.Cuenta) As Entidades.Cuenta
        Return crudCuenta.Retrieve(Of Entidades.Cuenta)
    End Function

    Friend Function Update(cuenta As Entidades.Cuenta)
        crudCuenta.Update(cuenta)
    End Function

    Friend Function Delete(cuenta As Entidades.Cuenta)
        crudCuenta.Delete(cuenta)
    End Function

End Class

File: Management/CuentaManagement.cs
using AccesoDatos.Crud;
using Entidades;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Management
{
    public class CuentaManagement
    {
        private CuentaCrudFactory crudCuenta;

        public CuentaManagement()
        {
            crudCuenta = new CuentaCrudFactory();
        }

        public void Create(Cuenta cuenta)
        {

            crudCuenta.Create(cuenta);

        }


        public List<Cuenta> RetrieveAll()
        {
            return crudCuenta.RetrieveAll<Cuenta>();
        }

        public Cuenta RetrieveById(Cuenta cuenta)
        {
            return crudCuenta.Retrieve<Cuenta>(cuenta);
        }

        public void Update(Cuenta cuenta)
        {
            crudCuenta.Update(cuenta);
        }

        public void Delete(Cuenta cuenta)
        {
            crudCuenta.Delete(cuenta);
        }
    }
}
